ivy_tests/test_ivy/test_functional/test_nn/test_losses.py

```diff
@@ -52,10 +52,10 @@ def test_cross_entropy(
     fw,
 ):
     pred_dtype, pred = dtype_and_pred
-    if fw == "torch" and pred_dtype == "float16":
-        return
     true_dtype, true = dtype_and_true
     length = min(len(true), len(pred))
+    true = true[:length]
```

Comment:
we should actually ensure that both arrays are the same length at _generation_ time, by using [st.shared](https://hypothesis.readthedocs.io/en/latest/data.html?highlight=shared#hypothesis.strategies.shared). See the [test](https://github.com/unifyai/ivy/blob/bdc34708852f87e4a600808d8892d0ddd3edc053/ivy_tests/test_ivy/test_functional/test_core/test_manipulation.py#L38) for `ivy.concat` for an example where there are shape consistency requirements for the arrays being tested, which we handle at data generation time.